### ISSUE TITLE
Add Kiwix reader to hotspot

### ISSUE THREAD

**Issue by rgaudin** [offspot/cardshop#336]
Now that ZIM files are downloadable from the hotspot, we've discussed including Kiwix readers into images built from Imager.
We need to answer some questions:

- always or chosen by Imager user?
- Which platform to include? And which file for those that have multiple variant (linux flatpack/appimage)
- Which version to use and update strategy? Should this be automatically the latest version at time of order? At time of configuration update? Should it be a manually selected version (implying we'd have someone validate that a version is good to go)?

**Comment by Popolechien** [comment]
Since as we've been saving quite a bit of space with the new iteration I think having it always included won't bother much and would prove to be a nice Easter egg of sorts.

Platforms to include: android and windows/ linux, which are the only ones we support officially (and maybe the PWA?) As for the linux flavours: the simplest package, and Appimage in that regard seems to be more portable, but generally speaking I would oppose having too wide a choice.

Use the latest version at the time of order. Since we're not really asking the user whether they want it or not in the first place, ease of use should prevail (ie the user should not have to think about anything).

**Comment by rgaudin** [comment]
> ease of use should prevail (ie the user should not have to think about anything).

I agree. Should I have been misunderstood, I meant that the included version could be a version we (cardshop maintainer) set and that applies to all cards. That would be to prevent new reader release going into images without having been somewhat tested by the cardshop team. That's what we would do with any external software (pinning a version). Not doing it because it's developed by Kiwix is up to discussion I think.

**Comment by kelson42** [comment]
I propose to move this request to offspot/package-requests.

Once created, and only once created, the discussion of default integration wull makes sense. Therefore i would recommend to have an other ticket for that.

IMO, this topic is intetestibg but might be more subtil than it looks like at first sight. Just an example: this package will be a few hundreds of MB big, do we really want to impose it if the user is short on nass storage? if the mass stirage is obly a few GB? If the user has no ZIM configured?

**Comment by rgaudin** [comment]
Makes sense.


> always or chosen by Imager user?

This somewhat answers it. We could include it as long as there's one ZIM file but then the first added ZIM would add more storage than displayed. We'd have to adapt the UI to explain what's going on. Propbably better then to have an option.

> Which platform to include? And which file for those that have multiple variant (linux flatpack/appimage)

@kelson42 can you give your feedback? Is what @Popolechien proposed OK with you? That will feed the package-requests ticket.

> Which version to use and update strategy? Should this be automatically the latest version at time of order? At time of configuration update? Should it be a manually selected version (implying we'd have someone validate that a version is good to go)?

Implementing is as a package request kind of settles it.

**Comment by kelson42** [comment]
>> Which platform to include? And which file for those that have multiple variant (linux flatpack/appimage)

> @kelson42 can you give your feedback? Is what @Popolechien proposed OK with you? That will feed the package-requests ticket.

You won't like my answer... but is that a different question than the one we have for the similar page on www.kiwix.org? I would solve it there and then do the package following that model (modulo links to app stores obviously).

**Comment by rgaudin** [comment]
Then I'll let you open appropriate tickets when you're ready.
